packages/astro/src/core/middleware/index.ts

```diff
@@ -1,9 +1,40 @@
-import type { MiddlewareResponseHandler } from '../../@types/astro';
+import type { MiddlewareResponseHandler, Params } from '../../@types/astro';
 import { sequence } from './sequence.js';
+import { createAPIContext } from '../endpoint/index.js';
 
 function defineMiddleware(fn: MiddlewareResponseHandler) {
 	return fn;
 }
 
+/**
+ * Payload for creating a context to be passed to Astro middleware
+ */
+export type CreateContext = {
+	/**
+	 * The incoming request
+	 */
+	request: Request;
+	/**
+	 * Optional parameters
+	 */
+	params?: Params;
+};
+
+/**
+ * Creates a context to be passed to Astro middleware `onRequest` function.
+ *
+ * @param request
+ * @param params
+ * @param adapterName
```

Comment:
I think the `@param`s here can be removed, since there's only a single object param